libstorage/drivers/storage/rbd/utils/utils.go

```diff
@@ -177,6 +175,34 @@ func GetMappedRBDs(ctx types.Context) (map[string]string, error) {
 	return devMap, nil
 }
 
+//ParseMappedRBDs is a two-stage parse attempt because Ceph v13 has different JSON format to previous versions
+func ParseMappedRBDs(output *[]byte) (map[string]*rbdMappedEntry, error) {
+	rbdMap := map[string]*rbdMappedEntry{}
+
+	err := json.Unmarshal(*output, &rbdMap)
+	if err != nil {
+		type v13RbdMappedEntry struct {
+			ID     string `json:"id"`
+			Device string `json:"device"`
+			Name   string `json:"name"`
+			Pool   string `json:"pool"`
+			Snap   string `json:"snap"`
+		}
+		rbdMapv2 := []*v13RbdMappedEntry{}
+		err2 := json.Unmarshal(*output, &rbdMapv2)
+		if err2 != nil {
+			return nil, goof.WithError(
+				"unable to parse rbd showmapped", err2)
+		}
+		for _, entry := range rbdMapv2 {
+			tmp := rbdMappedEntry{Device: entry.Device, Name: entry.Name, Pool: entry.Pool, Snap: entry.Snap}
+			rbdMap[entry.ID] = &tmp
```

Comment:
the `tmp` var isn't really necessary here. Just do `rbdMap[entry.ID] = &rbdMappedEntry{...}`